core/storage/question/gae_models.py

```diff
@@ -261,19 +261,18 @@ def get_question_skill_links_by_skill_ids(
         return question_skill_link_models, next_cursor_str
 
     @classmethod
-    def get_question_skill_links_based_on_mastery_equidistributed_by_skill(
-            cls, total_question_count, skill_ids, degrees_of_mastery):
+    def get_question_skill_links_based_on_difficulty_equidistributed_by_skill(
+            cls, total_question_count, skill_ids, difficulty_requested):
         """Fetches the list of constant number of QuestionSkillLinkModels
-        linked to the skills, sorted by the difference between skill
-        difficulty and skill mastery.
+        linked to the skills, sorted by the absolute value of the difference
+        between skill difficulty and the requested difficulty.
 
         Args:
             total_question_count: int. The number of questions expected.
             skill_ids: list(str). The ids of skills for which the linked
                 question ids are to be retrieved.
-            degrees_of_mastery: dict(str, float). The keys are the requested
-                skill IDs. The values are the corresponding mastery degree of
-                the user.
+            difficulty_requested: int. The number of the difficulty requested
```

Comment:
Sorry, what do you mean by this? I don't understand what "The number of the difficulty" is.

Also, I thought difficulties were floats between 0 and 1? If so, why is this an int?

Reply:
Done.